### DISCUSSION TITLE
Stelo connection auth failing

### DISCUSSION THREAD

**Discussion by jcbinf** [NightscoutFoundation/xDrip#3748] [Bug reports]
I am seeing failure to authenticate with Stelo sensor.  I have tried sept 19 release and latest nightly plus a  local build from source.  I have loaded the keks from the QR code using autoconfigure, checked setting per https://navid200.github.io/xDrip/docs/Dexcom/G7.html.  Still digging but out of my depth a bit still.  Wondering where do those keks keys come from? are they valid? Any suggestions appreciated. See the logcat snip below of failure on my vsmart 341u, android 10 (yes I know it is old and cheap).

11-02 11:25:11.008  7309  8246 I System.out: KEKS-Plugin :: Updating saved key from loaded persistence data
11-02 11:25:11.010  7309  8246 D Ob1G5CollectionService: Changing state from: DISCOVER to CHECK_AUTH
11-02 11:25:11.066  7309  7329 D BluetoothGatt: onConnectionUpdated() - Device=CA:1A:33:26:FA:52 interval=39 latency=0 timeout=500 status=0
11-02 11:25:11.118  7309 19520 D Ob1G5CollectionService: Status: Authorizing 2 11:25
11-02 11:25:11.121  7309 19520 I System.out: KEKS-Plugin :: Changing state: Unknown - > Round Start
11-02 11:25:11.238  7309  8246 D BluetoothGatt: setCharacteristicNotification() - uuid: f8083538-849e-531c-c594-30f1f86a4ea5 enable: true
11-02 11:25:11.240  7309  8246 D Ob1G5StateMachine: Extra data notifications enabled
11-02 11:25:11.336  7309  8246 D BluetoothGatt: setCharacteristicNotification() - uuid: f8083535-849e-531c-c594-30f1f86a4ea5 enable: true
11-02 11:25:11.338  7309  8246 I System.out: KEKS-Plugin :: Processing anext in state: Round Start
11-02 11:25:11.339  7309  8246 D Ob1G5StateMachine: Sending auth command: 
11-02 11:25:11.339  7309  8246 D Ob1G5StateMachine: 0x00000000 02 BD 81 7D D0 70 E0 D1 AD 02                   ...}.p....
11-02 11:25:11.438  7309  8246 D Ob1G5StateMachine: Received Authentication 2 indication bytes: 03E64488E11E526EBED3F6295CEC7E96B1
11-02 11:25:11.438  7309  8246 I System.out: KEKS-Plugin :: Received remote response: 03E64488E11E526EBED3F6295CEC7E96B1 when in Request Auth
11-02 11:25:11.439  7309  8246 I System.out: KEKS-Plugin :: Processing anext in state: Request Auth
11-02 11:25:11.440  7309  8246 D Ob1G5StateMachine: Sending auth command: 
11-02 11:25:11.440  7309  8246 D Ob1G5StateMachine: 0x00000000 04 BC 22 98 36 2D 15 42 F2                      ..".6-.B.
11-02 11:25:11.532  7309  8246 D Ob1G5StateMachine: Received Authentication 2 indication bytes: 050201
11-02 11:25:11.532  7309  8246 I System.out: KEKS-Plugin :: Received remote response: 050201 when in Challenge Reply
11-02 11:25:11.532  7309  8246 I System.out: KEKS-Plugin :: Could not authenticate!
11-02 11:25:11.532  7309  8246 I System.out: KEKS-Plugin :: Changing state: Challenge Reply - > Unknown
11-02 11:25:11.532  7309  8246 I System.out: KEKS-Plugin :: Processing anext in state: Unknown
11-02 11:25:11.532  7309  8246 D Ob1G5StateMachine: Null value returned via plugin
11-02 11:25:11.532  7309  8246 D Ob1G5StateMachine: Attempting state reset
11-02 11:25:11.532  7309  8246 D Ob1G5CollectionService: Resetting internal state by request
11-02 11:25:14.807  7309  7329 D BluetoothGatt: onClientConnectionState() - status=19 clientIf=5 device=CA:1A:33:26:FA:52
11-02 11:25:14.809  7309  7329 D Ob1G5CollectionService: Connection Disconnected/Failed: com.polidea.rxandroidble2.exceptions.BleDisconnectedException: Disconnected from MAC='XX:XX:XX:XX:XX:XX' with status 19 (GATT_CONN_TERMINATE_PEER_USER)
11-02 11:25:14.814  7309  7395 D Ob1G5CollectionService: Bluetooth connection: Disconnecting

**Comment by Navid200** [comment]
Post a screenshot of the status page.

**Reply by jcbinf** [reply]
After grabbing one screenshot, while checking it on the phone, the Stelo sensor spontaneosly paired with chimes and everything.  I did not instigate the action myself, and did not interact with anything other than the Files app.  I made another screenshot immediately; thus 2 posted.
I got an entire 24 hours data downloaded from the sensor, and for now the app appears to operate as expected.
This after 3  days of uncounted restarts, setting experiments, installing variuous builds...
Perhaps something is flaky with pairing on my device?
I am going to continue trying to figure it out.
![Screenshot_20241102-155350](https://github.com/user-attachments/assets/aad603ce-16ba-4c6d-aec9-079cda76b6e9)
![Screenshot_20241102-155608](https://github.com/user-attachments/assets/96080333-8003-4d24-b3dc-4e86b1cf93db)
